cmd/sealer/cmd/run.go

```diff
@@ -76,12 +89,66 @@ create a cluster with custom environment variables:
 		}
 
 		cluster := cf.GetCluster()
+
 		infraDriver, err := infradriver.NewInfraDriver(&cluster)
 		if err != nil {
 			return err
 		}
 
-		//TODO mount image and copy to cluster
+		hostsPlatformMap, err := infraDriver.GetHostsPlatform(infraDriver.GetHostIPList())
+		if err != nil {
+			return err
+		}
+
+		distributor, err := imagedistributor.NewScpDistributor(infraDriver)
+		if err != nil {
+			return err
+		}
+
+		imageEngine, err := imageengine.NewImageEngine(imagecommon.EngineGlobalConfigurations{})
+		if err != nil {
+			return err
+		}
+
+		var (
+			clusterImageName  = cluster.Spec.Image
+			rootfs            = infraDriver.GetClusterRootfs()
+			rootfsRegistryDir = filepath.Join(rootfs, "registry")
+		)
+
+		for platform, hosts := range hostsPlatformMap {
+			var (
+				mountDir         = filepath.Join(common.DefaultSealerDataDir, cluster.Name, "mount", platform.ToString())
+				mountRegistryDir = filepath.Join(mountDir, "registry")
+			)
+
+			// pull cluster image via it`s platform
+			if err := pullClusterImage(clusterImageName, platform, imageEngine); err != nil {
+				return err
+			}
+
+			// mount cluster image
+			if err := mountClusterImage(clusterImageName, mountDir, imageEngine); err != nil {
+				return err
+			}
+
+			// send registry to local host dir.
```

Comment:
how to send registry dir, should be implemented by RegistryConfigurator, imageEngine only needs to be responsible for preparing the registry directory.